contributors/devel/conformance-tests.md

```diff
@@ -1,57 +1,69 @@
-Once the new tests prove to be stable in CI runs, later create a follow up PR 
-to add the test to conformance. This approach also decouples the development 
-of useful tests from their promotion to conformance.
+The process to add new conformance tests is intended to decouple the
+development of useful tests from their promotion to conformance:
+
+- Contributors write and submit e2e tests, to be approved by owning SIGs
+- Tests are proven to meet the conformance test requirements
+- A follow up PR is submitted to promote the PR to conformance
 
-A conformance test verifies the expected functionality works as a user might encounter it in the wild, 
 and tests should begin by covering the most important and visible aspects of the function.
 
 ### Conformance Test Requirements
 
-A test is eligible for promotion to conformance if it meets the following requirements:
-
-- testing GA feature (not alpha or beta APIs, nor deprecated features)
-- must be portable (not dependent on provider-specific capabilities or on the public internet)
-- cannot test a feature which obviously cannot be supported on a broad range of platforms 
-(i.e. testing of multiple disk mounts, GPUs, high density)
-- cannot test an optional feature (e.g. not policy enforcement)
-- should be non-privileged (neither root on nodes, network, nor cluster)
-- cannot rely on any particular non-standard file system permissions granted to containers or users 
-(i.e. sharing writable host /tmp with a container)
-- should be stable and run consistently
-- cannot skip providers (there should be no Skip like directives added to the test), 
-especially in the Nucleus or Application layers as described 
-[here](https://github.com/kubernetes/community/blob/master/contributors/devel/architectural-roadmap.md).
-- cannot test cloud provider specific features (i.e. GCE monitoring, S3 Bucketing, ...)
-- should work with default settings for all configuration parameters 
-(example: the default list of admission plugins should not have to be tweaked for passing conformance).
-- cannot rely on any binaries that are not required for the
-linux kernel or for a kubelet to run (i.e. git)
+Conformance tests should be boring. They should rely on nothing special, test only
+features that are available everywhere, and generate the same results every time.
+
+More specifically, a test is eligble for promotion to conformance if:
+
+- it tests only GA, non-optional features or APIs (ie: no alpha or beta endpoints, 
+  no feature flags required)
+- it works for all providers (ie: no `SkipIfProviderIs`/`SkipUnlessProviderIs` calls)
+- it is non-privileged (ie: no root on nodes, network, nor cluster)
+- it works without access to the public internet
+- it works without non-standard filesystem permissions granted to containers or users
+  (eg: can't share a writable host /tmp with a container)
+- it does not rely on any binaries that would be required for the linux kernel or
+  kubelet to run (ie: can't rely on git)
```

Comment:
Some of these constraints seem weirdly arbitrary and not really a requirement for some tests.

Reply:
Can we just abstract this a bit and talk about how it can't require platform, or vendor, specific artifacts? Being too specific here might lead to people wanting/needing to add more and more over time, and I don't think this should be the complete list, rather the guidelines.

Reply:
How so?  Trying to spell out explicit violations we've seen in the past

Reply:
Yep, I get the intent. I'm just not keen on having to list each and every way people may violate our goals here, and keep updating it as they find new ways :-)   So, if possible, let's talk about the idea in the abstract.  E.g., perhaps:
it works on all providers without any proprietary add-ons and without requiring a configuration that is not deemed acceptable (by SIG-Arch) for running the test suite.

The point of the last part is that I'm not sure we can enumerate all of the possible config options in this list, e.g. doing things like requiring sharable/write /tmp with the host might, or might not, be ok in the future. SIG-Arch should define what kind of setup in which the tests should run and what the requirements are on the environment in-which it runs. That doc can, and will, change over time and maybe per release so it should be versioned, but in this doc (which is more like our charter to me) should be written to be pretty static across all versions - if we can.

Reply:
This list needs to be specific enough that it can be used as a checklist for those reviewing whether or not a given test is eligible for promotion to conformance